ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/storage/keyvalue/DatabaseMetadata.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.storage.keyvalue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class DatabaseMetadata {
+  private static final String METADATA_FILENAME = "DATABASE_METADATA.json";
+  private static ObjectMapper MAPPER = new ObjectMapper();
+  private final int version;
+
+  @JsonCreator
+  DatabaseMetadata(@JsonProperty("version") final int version) {
+    this.version = version;
+  }
+
+  public int getVersion() {
+    return version;
+  }
+
+  static DatabaseMetadata fromDirectory(final Path databaseDir) throws IOException {
+    try {
+      return MAPPER.readValue(getDefaultMetadataFile(databaseDir), DatabaseMetadata.class);
```

Comment:
What about the case where the file exists but is corrupted?  I would prefer a total service failure instead of proceeding at version 0, which may be incorrect.

Reply:
When the file is corrupted, currently it's a failure with a jackson `UnrecognizedPropertyException` since I'm only catching `FileNotFoundException`. Do you want me to wrap it our own exception or is the `UnrecognizedPropertyException` ok?
```
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "🦄" (class tech.pegasys.pantheon.ethereum.storage.keyvalue.DatabaseMetadata), not marked as ignorable (one known property: "version"])
 at [Source: (File); line: 1, column: 11] (through reference chain: tech.pegasys.pantheon.ethereum.storage.keyvalue.DatabaseMetadata["🦄"])
```

Reply:
My main concern is that we don't continue at version 0 unless the file is truly not there.

Reply:
Yeah, if the file doesn't exist, then we continue at version 0. Otherwise we try to parse it. I added the test cases you mentioned in the other comments and will go ahead and wrap the exception.